x/evmutil/keeper/conversion_evm_native.go

```diff
@@ -102,13 +110,23 @@ func (k Keeper) ConvertERC20ToCoin(
 		return err
 	}
 
+	amountToLock := amount.BigInt()
+	amountToMint := amount.BigInt()
+
+	if isBep3Asset(pair.Denom) {
```

Comment:
I see what you mean know, and this is a nice approach showing there is are two paths in the logic: One for bep3 assets, and one for not.  Also, makes it clear that the amounts are only modified in the bep3 case, but also keeps the dirty details down a level in a function call.